docs: review documentation files (#787)
MobilityData/gtfs-validator: docs/ARCHITECTURE.md

> ### Data pipeline 📥➡️♨➡️📤

Reviewer: IMHO this document could use a few simple sentences at the top of the document describing how many of the classes for GTFS loading and validation are auto-generated using AutoValue and annotations. This is really the huge difference between v1 and v2 and it's somewhat buried in the bullets right now.

1️⃣ **Inputs**
- A local GTFS archive or fully qualified URL to download distance a GTFS archive from
- Command line arguments 

2️⃣ **Validator loading**
- Locate all validators and load them

3️⃣ **Feed loading**

> - Create `GtfsInput`

Reviewer: "- Create `GtfsInput`" isn't going to be meaningful to most people reading this document. I would eliminate it and just have 3 bullet points in this section.

  - Read GTFS files
  - Create `GtfsTableContainer` from data

> - Invoke and execute all `SingleEntityValidator`

Reviewer: ```suggestion
- Invoke and execute all `SingleEntityValidators` to validate data types, etc.
```

4️⃣ **Validators execution**

> - Invoke and execute all `FileValidator` in parallel

Reviewer: ```suggestion
- Invoke and execute all `FileValidators` in parallel to validate GTFS semantic rules
```

 
5️⃣ **Notice export**

> 1. Creates path to export notices as specified by command line input `--output` (or `-o`).

Reviewer: I think you can get rid of this first bullet point here in favor of the below

> 1. Export notices in `NoticeContainer`. Notices are alphabetically sorted, and stored by type sorted  `.json` file. 

Reviewer: ```suggestion
1. Export notices from `NoticeContainer` to two JSON files in the specified directory - `report.json` for validator results and `system_errors.json` for any software errors that occurred during validation. Notices are alphabetically sorted in the `.json` files. 
```

I collapsed the above bullet into this one. Also, is the 2nd sentence above what was intended? There is something wrong with it.

> 🔚 **Output: validation result and system errors reports** 

Reviewer: I'd remove this bullet point -- seems like a duplicate of above.
